docs/translations/translation-basics.md

```diff
@@ -0,0 +1,59 @@
+# Translation basics
```

Comment:
I'm not sure if it's fully worth it or not, but maybe also talking about how translation is shaped, it comes in 3 forms:
- A `.pot` file that contains all the original translation and paths to it, this file is unique to the plugin, for example: `woo-gutenberg-products-block.pot`
- A `.po` and `.mo` which pairs of strings and translation, this is unique to the plugin and language and handles only PHP strings, e.g. `woo-gutenberg-products-block-nl_NL.po`.
- JSON files, which are unique to the plugin langauge, and specific file, those have the [JED](https://messageformat.github.io/Jed/) format and are named as `{plugin_slug}-{language_code}-{md5(filepath)}.json`, for example `woo-gutenberg-products-block-nl_NL-df69ae21391b08cd8133e643dacbc72d.json` is the JSON translation file of `build/checkout-frontend.js` file in Dutch.

Reply:
Thanks for pointing out these cases, @senadir and @dinhtungdu. I added the new doc `docs/translations/translation-loading.md` where I've added these sections.